mani_skill/envs/tasks/control/hopper.py

```diff
@@ -0,0 +1,266 @@
+"""Adapted from https://github.com/google-deepmind/dm_control/blob/main/dm_control/suite/hopper.py"""
+
+import os
+from typing import Any, Dict, Union
+
+import numpy as np
+import torch
+from transforms3d.euler import euler2quat
+
+from mani_skill.agents.base_agent import BaseAgent
+from mani_skill.agents.controllers import *
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.envs.utils import randomization, rewards
+from mani_skill.sensors.camera import CameraConfig
+from mani_skill.utils import common, sapien_utils
+
+# fix new imports later
+from mani_skill.utils.building.ground import build_ground
+from mani_skill.utils.registration import register_env
+from mani_skill.utils.structs.pose import Pose
+from mani_skill.utils.structs.types import (
+    Array,
+    GPUMemoryConfig,
+    SceneConfig,
+    SimConfig,
+)
+
+MJCF_FILE = f"{os.path.join(os.path.dirname(__file__), 'assets/hopper.xml')}"
+
+# params from dm_control
+# Minimal height of torso over foot above which stand reward is 1.
+_STAND_HEIGHT = 0.6
+
+# Hopping speed above which hop reward is 1.
+_HOP_SPEED = 2
+
+
+class HopperRobot(BaseAgent):
+    uid = "hopper"
+    mjcf_path = MJCF_FILE
+
+    @property
+    def _sensor_configs(self):
+        return [
+            # replicated from xml file
+            CameraConfig(
+                uid="cam0",
+                pose=Pose.create_from_pq([0, -2.8, 0], euler2quat(0, 0, np.pi / 2)),
+                width=128,
+                height=128,
```

Comment:
is 128x128 correct? I recall dm-control uses smaller images

Reply:
i found that images are 240 height, 320 width:

```
import dm_control.suite as suite
from dm_control.suite.wrappers import pixels
import numpy as np

env = suite.load("hopper", "hop")
env_and_pixels = pixels.Wrapper(env, pixels_only=True)

_, _, _, obs = env_and_pixels.reset()

print(obs['pixels'].shape) # prints (240, 320, 3)
```
this seems large, should we use the same dims?

Reply:
Actually turns out people seem to use different resolutions. Let's just stick with 128x128 for now. It runs fast and is sufficient.

Just checked TDMPC2 code and they use 384x384. So not sure how this is picked usually.